### ISSUE TITLE
[Feature Request] Support functions written by KCL and its SDKs for KPT 

### ISSUE THREAD

**Issue by Peefy** [GoogleContainerTools/kpt#3671]
### Describe your problem

Please describe your problem and specific use case, providing as much context as
possible. As a general rule, it is more effective to focus on describing the
_problem_ instead of asking for a specific _solution_. This ensures the
following:

1. We understand _WHY_ you are asking for this.

I was very glad to find the KPT project and browse all its documents. I found that it coincided with what we wanted to do.

Specifically, we have just opened our domain language project [KCL](https://github.com/KusionStack/KCLVM), which is specifically used for configuration writing and policy verification in cloud-native and kubernetes scenarios. However, it is obviously not enough to use only one language for kubernetes configuration management until we find KPT. We want to develop KCL plug-ins and SDKs for KPT to write KPT functions, so we want to keep in touch with you through this issue. Because I am new to KPT, I want to know where I can get help documents if I want to develop a new language SDK for KPT.

2. We can pattern match across multiple users with this problem.
3. Allow design freedom for addressing the problem given various trade-offs and
   UX considerations.

**Comment by droot** [comment]
Thanks for reaching out @Peefy 

At the moment, we do not have documentation for writing a new language SDK for `kpt`. But you can take a look at our docs for [Developing Functons](https://kpt.dev/book/05-developing-functions/) and see the docs for the three SDK (Go, Typescript, Starlark) to get an idea how function authors use the SDKs. Hopefully that should get you started.

**Comment by Peefy** [comment]
> Thanks for reaching out @Peefy 
> 
> At the moment, we do not have documentation for writing a new language SDK for `kpt`. But you can take a look at our docs for [Developing Functons](https://kpt.dev/book/05-developing-functions/) and see the docs for the three SDK (Go, Typescript, Starlark) to get an idea how function authors use the SDKs. Hopefully that should get you started.
> 

Are there plans to support other languages? I will follow kpt for a long time. 😄😄

**Comment by droot** [comment]
> Are there plans to support other languages?

Not at the moment. we are stretched thin even to support the three we do currently. So contributors are welcome :)

**Comment by Peefy** [comment]
> > Are there plans to support other languages?
> 
> Not at the moment. we are stretched thin even to support the three we do currently. So contributors are welcome :)

Of course, I am very willing to contribute and participate in the kpt project. If I can, I will continue to pay attention to kpt and browse all documents and codes and give my design.

**Comment by droot** [comment]
> > > Are there plans to support other languages?
> > 
> > 
> > Not at the moment. we are stretched thin even to support the three we do currently. So contributors are welcome :)
> 
> Of course, I am very willing to contribute and participate in the kpt project. If I can, I will continue to pay attention to kpt and browse all documents and codes and give my design.

Great to hear!

**Comment by Peefy** [comment]
Hello. I have completed the [initial version of the kpt KCL function](https://github.com/KusionStack/kpt-kcl-sdk), and I hope you can give some suggestions. If possible, I am willing to contribute to the KPT community.

**Comment by natasha41575** [comment]
@yuwenma assigning to you as the area lead of function sdk

**Comment by Peefy** [comment]
Hi @droot. Long time no contact. I am excited to report on the progress to you. and we have completed the [KPT KCL Function PoC](https://kcl-lang.io/docs/user_docs/guides/working-with-k8s/kpt_kcl_sdk) and expected to contribute to the community, there is no additional demand for `kpt fn`.

**Comment by droot** [comment]
Sorry @Peefy for not responding earlier. I plan to review it next week. Will share the feedback then. Thanks for the update.

**Comment by kispaljr** [comment]
Hi @Peefy, the kpt project has new maintainers and we decided to try to revive the project. 
Would you be still interested to add KCL support to KRM function SDK?
